iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -66,34 +73,29 @@ PoshRuntime::~PoshRuntime() noexcept
 }
 
 
-const std::string& PoshRuntime::verifyInstanceName(const std::string& name) noexcept
+const ProcessName_t& PoshRuntime::verifyInstanceName(const ProcessName_t& name) noexcept
 {
     if (name.empty())
     {
         LogError() << "Cannot initialize runtime. Application name must not be empty!";
         std::terminate();
     }
-    else if (name.compare(DEFAULT_RUNTIME_INSTANCE_NAME) == 0)
+    else if (name.compare(defaultRuntimeInstanceName()) == 0)
     {
         LogError() << "Cannot initialize runtime. Application name has not been specified!";
         std::terminate();
     }
-    else if (name.front() != '/')
+    else if (name.c_str()[0] != '/')
```

Comment:
Magic slash? To understand the context better you could maybe create a variable
```cpp
constexpr char PATH_SEPARATOR = '/';
if ( name[0] != PATH_SEPARATOR ) ...
```
Then a reviewer maybe understands better why we require this.

Reply:
Maybe we should get rid of the slash. As far as I know it's removed when the unix domain socket is used. I also wouldn't call it `PATH_SEPARATOR` since it's the same on Windows, even with `\` as path separator. How about `APP_PREFIX`

Reply:
Were should this live? If locally it does not add much, just makes the code artificially longer. If globally is not really useful...unless we want to make it configurable somehow. This is not needed here I would say.

Reply:
I guess we have still the requirement that the name provided when creating the runtime has to start with a slash. And I guess the reason was that this string is used to name a message queue. As we heavily break this API now, we could clean this up, or?

Reply:
Maybe we should hide this from the user. If the name like for message queues has to start with a slash then the message queue adds one if no slash is present otherwise it leaves it as it is. 
It would also have no effect for the user since he would still find the corresponding file handle under the specified name.

Reply:
I've created #381 for this